python/activator/activator.py

```diff
@@ -269,6 +408,13 @@ def parse_next_visit(http_request):
     if not event.data:
         raise ValueError("empty CloudEvent received")
 
+    if platform == "knative":
+        # Calculate time to load knative and receive message based on time header from knative request
+        _log.info("Fan out send event at %s", event['time'])
+        fan_out_knative_msg_timestamp = float(event['time'])
+        fan_out_to_prompt_time = time.time() - fan_out_knative_msg_timestamp
+        _log.info("Seconds since fan out message delivered %r", fan_out_to_prompt_time)
```

Comment:
This code will crash in two circumstances:
- `event` is not actually a valid nextVisit event. This can be fixed by moving this block to after `visit` is constructed.
- Next-Visit Fan Out does not send a `time` header. Either we need to treat this as a breaking change in PP, or the code needs to be modified to handle events with no `time`. I'd prefer the latter, of course; it seems absurd to make a major release over some debugging code.

Reply:
Same comment as above about logging levels.

Reply:
Cloud Events have header attributes and data.  The nextVisit event is sent in the data payload of the Cloud Event.  Time is sent in the Cloud Event header as an attribute.   I will update the code to allow for time not being sent.